# Google Maps API 설치

- conda 명령어로 googlemaps 모듈 설치
	- conda install -c conda-forge googlemaps

In [4]:
import googlemaps

gmaps_key = ""
gmaps = googlemaps.Client(key=gmaps_key)

- 단순 테스트 코드

	-> 정상 설치되었다면 리스트로 위치 결과가 반환된다.

In [33]:
gmaps.geocode("영등포경찰서", language="ko")

[{'address_components': [{'long_name': '영등포구',
    'short_name': '영등포구',
    'types': ['political', 'sublocality', 'sublocality_level_1']},
   {'long_name': '서울특별시',
    'short_name': '서울특별시',
    'types': ['administrative_area_level_1', 'political']},
   {'long_name': '대한민국',
    'short_name': 'KR',
    'types': ['country', 'political']}],
  'formatted_address': '대한민국 서울특별시 영등포구',
  'geometry': {'bounds': {'northeast': {'lat': 37.556286, 'lng': 126.9498867},
    'southwest': {'lat': 37.4850011, 'lng': 126.8779285}},
   'location': {'lat': 37.5223245, 'lng': 126.9101692},
   'location_type': 'APPROXIMATE',
   'viewport': {'northeast': {'lat': 37.556286, 'lng': 126.9498867},
    'southwest': {'lat': 37.4850011, 'lng': 126.8779285}}},
  'partial_match': True,
  'place_id': 'ChIJk_FAvx2ffDURQAuECsjy0Cw',
  'types': ['political', 'sublocality', 'sublocality_level_1']}]

#### 🔰 구글맵 API에서 데이터 얻기

- `geocode()` 함수는 전체 결과 크기가 1인 리스트 자료형을 반환한다.

- 리스트 안에 하나의 큰 dict형으로 정보가 담겨있다.
- index [0]으로 한 개밖에 없는 dict형 요소에 접근할 수 있다.
- dict형에서 데이터를 얻는 get(__key) 메서드를 사용해 정보에 접근하다.

In [34]:
temp = gmaps.geocode("서울영등포경찰서", language="ko")
temp

[{'address_components': [{'long_name': '608',
    'short_name': '608',
    'types': ['premise']},
   {'long_name': '국회대로',
    'short_name': '국회대로',
    'types': ['political', 'sublocality', 'sublocality_level_4']},
   {'long_name': '영등포구',
    'short_name': '영등포구',
    'types': ['political', 'sublocality', 'sublocality_level_1']},
   {'long_name': '서울특별시',
    'short_name': '서울특별시',
    'types': ['administrative_area_level_1', 'political']},
   {'long_name': '대한민국',
    'short_name': 'KR',
    'types': ['country', 'political']},
   {'long_name': '150-043',
    'short_name': '150-043',
    'types': ['postal_code']}],
  'formatted_address': '대한민국 서울특별시 영등포구 국회대로 608',
  'geometry': {'location': {'lat': 37.5260441, 'lng': 126.9008091},
   'location_type': 'ROOFTOP',
   'viewport': {'northeast': {'lat': 37.5273930802915,
     'lng': 126.9021580802915},
    'southwest': {'lat': 37.5246951197085, 'lng': 126.8994601197085}}},
  'partial_match': True,
  'place_id': 'ChIJ1TimJLaffDURptXOs0Tj6s

In [35]:
len(temp)

1

In [36]:
len(temp[0])

7

In [37]:
temp[0].keys()

dict_keys(['address_components', 'formatted_address', 'geometry', 'partial_match', 'place_id', 'plus_code', 'types'])

- 필요한 정보를 가져온다.
- 경찰서가 위치한 구 이름과 경찰서의 지도 상 위치 정보가 필요하다.

	- 'formatted_address': '대한민국 서울특별시 영등포구 국회대로 608',</br>
	'geometry': {'location': {'lat': 37.5260441, 'lng': 126.9008091}, ... }

In [38]:
temp[0].get("geometry")

{'location': {'lat': 37.5260441, 'lng': 126.9008091},
 'location_type': 'ROOFTOP',
 'viewport': {'northeast': {'lat': 37.5273930802915, 'lng': 126.9021580802915},
  'southwest': {'lat': 37.5246951197085, 'lng': 126.8994601197085}}}

In [39]:
temp[0].get("geometry")["location"]

{'lat': 37.5260441, 'lng': 126.9008091}

In [40]:
type(temp[0].get("geometry")["location"])

dict

In [41]:
print(
	temp[0].get("geometry")["location"]["lat"], # latitude, 위도
	temp[0].get("geometry")["location"]["lng"] # longitude, 경도
)

37.5260441 126.9008091


In [42]:
temp[0].get("formatted_address")

'대한민국 서울특별시 영등포구 국회대로 608'

In [43]:
temp[0].get("formatted_address").split()

['대한민국', '서울특별시', '영등포구', '국회대로', '608']

In [44]:
# 전체 주소에서 필요한 구 이름만 가져온다.

temp[0].get("formatted_address").split()[2]

'영등포구'